### PR TITLE
[RFR] Add extras resolveVariables filter documentation

### PR DESCRIPTION
**Pull request by pascalduez**
Following of:

https://github.com/SassDoc/sassdoc-theme-default/issues/78
https://github.com/SassDoc/sassdoc-extras/pull/19
https://github.com/SassDoc/sassdoc-theme-default/pull/80

### CODE REVIEW

File: extra-tools/index.md [hunk 1]
```diff
@@ -147,3 +147,80 @@ The `description` key contains raw description text. The `descriptionPath` conta
...
+
+## Resolved variables (`resolveVariables`)
+
+The `resolveVariables` filter attempts to connect aliased variable to their
+original value. This might be useful for previewing certain types of values,
+e.g. colors. Variable aliases can be used in different contexts: variables, map properties, mixin and function parameters. While the original value is left
+untouched there's a new `resolvedValue` key being added.
+
+{% highlight scss %}
+// Blue
```
Comment (HugoGiraudel) [inline extra-tools/index.md]: Triple slash. Same further.

File: extra-tools/index.md [hunk 2]
```diff
@@ -147,3 +147,80 @@ The `description` key contains raw description text. The `descriptionPath` conta
...
 
 <p class="note  note--info"><strong>Note:</strong>  If the `markdown` filter is called **after** the `description` filter, it will parse the description value as Markdown.</p>
+
+
+## Resolved variables (`resolveVariables`)
+
+The `resolveVariables` filter attempts to connect aliased variable to their
+original value. This might be useful for previewing certain types of values,
+e.g. colors. Variable aliases can be used in different contexts: variables, map properties, mixin and function parameters. While the original value is left
+untouched there's a new `resolvedValue` key being added.
```
Comment (HugoGiraudel) [inline extra-tools/index.md]: Unnecessary line break?

File: extra-tools/index.md [hunk 3]
```diff
@@ -147,3 +147,80 @@ The `description` key contains raw description text. The `descriptionPath` conta
...
+$colors: (
+  'main': $color-blue,
+);
+
+/// Example mixin.
+/// @param {Color} $color [$color-main] - configurable background color
+@mixin foobar($color: $color-main) { ... }
+{% endhighlight %}
+
+Example: default items data
```
Comment (HugoGiraudel) [inline extra-tools/index.md]: Missing full stop. Or better, make a real sentence out of this. :)

File: extra-tools/index.md [hunk 4]
```diff
@@ -147,3 +147,80 @@ The `description` key contains raw description text. The `descriptionPath` conta
...
+  "context": {
+    "type": "variable",
+    "name": "color-main",
+    "value": "$color-blue"
+  },
+  "type": "Color"
+}
+{% endhighlight %}
+
+Example: after using the `resolveVariables` filter
```
Comment (HugoGiraudel) [inline extra-tools/index.md]: Missing full stop. Or better, make a real sentence out of this. :)

Comment (pascalduez) [inline extra-tools/index.md]: Well, this uses the recently updated style.

**Comment by pascalduez** [comment]
Fixed review points

**Comment by pascalduez** [comment]
Last call @SassDoc/owners ?

File: extra-tools/index.md [hunk 5]
```diff
@@ -147,3 +147,79 @@ The `description` key contains raw description text. The `descriptionPath` conta
 The `descriptionPath` is relative to the configuration file, and has no required format.
 
 <p class="note  note--info"><strong>Note:</strong>  If the `markdown` filter is called **after** the `description` filter, it will parse the description value as Markdown.</p>
+
+
+## Resolved variables (`resolveVariables`)
+
+The `resolveVariables` filter attempts to connect aliased variable to their
+original value. This might be useful for previewing certain types of values,
+e.g. colors. Variable aliases can be used in different contexts: variables, map properties, mixin and function parameters. While the original value is left untouched there's a new `resolvedValue` key being added.
```
Comment (HugoGiraudel) [inline extra-tools/index.md]: *there’s*

File: extra-tools/index.md [hunk 6]
```diff
@@ -147,3 +147,79 @@ The `description` key contains raw description text. The `descriptionPath` conta
...
+$colors: (
+  'main': $color-blue,
+);
+
+/// Example mixin.
+/// @param {Color} $color [$color-main] - configurable background color
+@mixin foobar($color: $color-main) { ... }
+{% endhighlight %}
+
+Example: default data for a variable item:
```
Comment (HugoGiraudel) [inline extra-tools/index.md]: Double colon, not a fan. :(

File: extra-tools/index.md [hunk 7]
```diff
@@ -147,3 +147,79 @@ The `description` key contains raw description text. The `descriptionPath` conta
...
+  "context": {
+    "type": "variable",
+    "name": "color-main",
+    "value": "$color-blue"
+  },
+  "type": "Color"
+}
+{% endhighlight %}
+
+Example: result after using the `resolveVariables` filter:
```
Comment (HugoGiraudel) [inline extra-tools/index.md]: Double colon, not a fan. :(

File: extra-tools/index.md (hunk 6 above)
Comment (pascalduez) [inline extra-tools/index.md]: Me neither actually, it's just to be consistent with the previous PR that introduced them.

So would `Example: default data for a variable item.` be better ?

File: extra-tools/index.md (hunk 5 above)
Comment (pascalduez) [inline extra-tools/index.md]: You mean correct typographic single quote ?

Comment (HugoGiraudel) [inline extra-tools/index.md]: Yes.

File: extra-tools/index.md (hunk 6 above)
Comment (HugoGiraudel) [inline extra-tools/index.md]: Yep.

**Pull request merged**